datasets.py

```diff
@@ -31,47 +38,188 @@ def create_mnist_datasets(
     test_dataset = tf.data.Dataset.from_tensor_slices(test_images).shuffle(test_size).batch(batch_size)
     return train_dataset, test_dataset
 
+'''
+Spoken digit audio dataset
+'''
+
+def _preprocess_spoken_digit_audio_tf(example, with_labels):
+    audio = example['audio']
+    # convert from int16? to float32 [-1.0, 1.0]
+    if audio.dtype.is_integer:
+        # detected clipping using just tf.int16.max to normalize...
+        audio = tf.cast(audio, tf.float32) / (tf.int16.max + 1.0)
+        
+    # remove DC offset
+    audio_mean = tf.math.reduce_mean(audio, axis=0) 
+    audio -= audio_mean
+    
+    if with_labels:
+        return (audio, example['label'])
+    return audio
+
+
+def _load_spoken_digit_dataset_tf(percent_load, load_labels, download=False):
+    # this dataset only has 'train' split
+    all_data = tfds.load(name='spoken_digit',
+                         split=[f'train[:{percent_load}%]'],
+                         shuffle_files=False,
+                         as_supervised=False,
+                         download=download)
+
+    all_data = all_data[0]
+    all_data = all_data.map(lambda x: _preprocess_spoken_digit_audio_tf(x, load_labels), 
+                            num_parallel_calls=tf.data.AUTOTUNE,
+                            deterministic=False)
+    return all_data
+
+
+def convert_raw_spoken_digit_dataset(destination_dir, percent_load=100, download=False):
```

Comment:
Why is `download` set to `False` by default?

Reply:
My reasoning is that the user should be forced to do that consciously since it starts a long process / downloads a large file to their system

That said, once it's downloaded I don't think it re-downloads if it finds it already residing in the expected location, so it's not a huge deal either way.